parse/parser.go

```diff
@@ -381,6 +381,77 @@ func parseStringLiteral(buf TokenBuffer) (ast.Expression, error) {
 	return &ast.StringLiteral{Value: token.Val}, nil
 }
 
+func parseFunctionLiteral(buf TokenBuffer) (*ast.FunctionLiteral, error) {
+	lit := &ast.FunctionLiteral{}
+	var err error
+
+	if err = expectNext(buf, Function); err != nil {
+		return nil, err
+	}
+
+	tok := buf.Read()
+	if tok.Type != Ident {
+		return nil, parseError{
+			tok.Type,
+			"token is not identifier",
+		}
+	}
+	lit.Name = &ast.Identifier{Value: tok.Val}
+
+	if err = expectNext(buf, Lparen); err != nil {
+		return nil, err
+	}
+
+	if lit.Parameters, err = parseFunctionParameters(buf); err != nil {
+		return nil, err
+	}
+
+	if err := expectNext(buf, Lbrace); err != nil {
+		return nil, err
+	}
+
+	if lit.Body, err = parseBlockStatement(buf); err != nil {
+		return nil, err
+	}
+	return lit, nil
+}
+
+func parseFunctionParameters(buf TokenBuffer) ([]*ast.ParameterLiteral, error) {
```

Comment:
plz add comment of this function

Reply:
when parameter has only ident , ex) func example(a a)
this function didn't return error.
I think it's a good idea to check error in parseFunctionParameters.

Reply:
Ok